azurerm/resource_arm_analysis_services_server.go

```diff
@@ -0,0 +1,406 @@
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	name := id.Path["servers"]
+
+	server, err := client.GetDetails(ctx, resourceGroup, name)
+
+	if err != nil {
+		if utils.ResponseWasNotFound(server.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Azure Analysis Services Server %q: %+v", name, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+
+	if location := server.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	d.Set("sku", *server.Sku.Name)
+
+	if serverProps := server.ServerProperties; serverProps != nil {
+		if serverProps.AsAdministrators == nil || serverProps.AsAdministrators.Members == nil {
+			d.Set("admin_users", []string{})
+		} else {
+			d.Set("admin_users", *serverProps.AsAdministrators.Members)
+		}
+
+		enablePowerBi, fwRules := flattenAnalysisServicesServerFirewallSettings(serverProps)
+		d.Set("enable_power_bi_service", enablePowerBi)
+		d.Set("ipv4_firewall_rule", fwRules)
+
+		d.Set("querypool_connection_mode", string(serverProps.QuerypoolConnectionMode))
+	}
+
+	flattenAndSetTags(d, server.Tags)
+
+	return nil
+}
+
+func resourceArmAnalysisServicesServerUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).analysisServicesServerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure ARM Analysis Services Server creation.")
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		server, err := client.GetDetails(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(server.Response) {
+				return fmt.Errorf("Error checking for presence of existing Analysis Services Server %q (Resource Group %q): %s", name, resourceGroup, err)
+			}
+		}
+
+		if server.ID != nil && *server.ID != "" {
+			return tf.ImportAsExistsError("azurerm_analysis_services_server", *server.ID)
+		}
+	}
+
+	sku := d.Get("sku").(string)
+
+	serverProperties := expandAnalysisServicesServerMutableProperties(d)
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	analysisServicesServer := analysisservices.ServerUpdateParameters{
+		Sku:                     &analysisservices.ResourceSku{Name: &sku},
+		Tags:                    expandTags(tags),
+		ServerMutableProperties: serverProperties,
+	}
+
+	future, err := client.Update(ctx, resourceGroup, name, analysisServicesServer)
+	if err != nil {
+		return fmt.Errorf("Error creating Analysis Services Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for completion of Analysis Services Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	server, err := client.GetDetails(ctx, resourceGroup, name)
+	if err != nil {
+		return err
```

Comment:
could we wrap this error to provide a better explanation to users if this fails?

```suggestion
		return fmt.Errorf("Error retrieving Analysis Services Server %q (Resource Group %q): %s", name, resourceGroup, err)
```